File: csharp/ex082.cs
// Faça um algoritmo que leia a nota de 10 alunos de uma turma e guarde-as em
// um vetor. No final, mostre:
// a) Qual é a média da turma
// b) Quantos alunos estão acima da média da turma
// c) Qual foi a maior nota digitada
// d) Em que posições a maior nota aparece

class Ex082
{
  public int[] vector  = new int[10];

  public int countNotes;
  public int avgGradeStudents;

  public int avgMajorStudents;
  public int majorGrade = 0;



  public void FirstCountdown()
  {
    for (int i = 0; i < vector.Length; i++)
    {
      Console.WriteLine($"Digite a nota do aluno: {i}");
      vector[i] = Convert.ToInt32(Console.ReadLine());

      countNotes += 1;

      avgGradeStudents = countNotes / vector.Length;
      if (vector[i] > avgGradeStudents)
      {
        avgMajorStudents += 1;
      }
    }

    Console.WriteLine($"Media da turma: {avgGradeStudents}");
    Console.WriteLine($"Quantos alunos estao acima da média da turma: {avgMajorStudents}");

    
    for (int j = 0; j < vector.Length; j++)
    {
      if (vector[j] > majorGrade)
      {
        Console.WriteLine($"Maior nota digitada: {majorGrade}");
        majorGrade = vector[j];
        Console.Write($"Posiçoes em que a maior nota aparece: [{j}]");
      }
    }



  }
}

File: javascript/ex082.js
// Faça um algoritmo que leia a nota de 10 alunos de uma turma e guarde-as em
// um vetor. No final, mostre:
// a) Qual é a média da turma
// b) Quantos alunos estão acima da média da turma
// c) Qual foi a maior nota digitada
// d) Em que posições a maior nota aparece

let arrNotas = [];
let acimaDaMedia = 0;
let maioresNotas = [];
let media = 0;
let totNotas = 0;
let maiorNota = 0;

var readline = require('readline-sync');


for (let i = 0; i <=9; i++) {
    arrNotas[i] = parseFloat(readline.question(`Pessoa n${i}, Digite sua Nota: `));
    
    totNotas+= arrNotas[i];
    
    media = totNotas/arrNotas.length;

    if (arrNotas[i] > maiorNota) {
        maiorNota = arrNotas[i];
    }

    
}

for (let i = 0; i < arrNotas.length; i++) {
    if (arrNotas[i] > media) {
        acimaDaMedia++;
    }
    
    if (arrNotas[i] === maiorNota) {
        maioresNotas[i] = i;
    }
    
}

 

console.log(`\n // a) Qual é a média da turma: ${media}
// b) Quantos alunos estão acima da média da turma: ${acimaDaMedia}
// c) Qual foi a maior nota digitada: ${maiorNota}
// d) Em que posições a maior nota aparece: [${maioresNotas}]`);






